### PR TITLE
Update supported Python and Django versions

### PR DESCRIPTION
**Pull request by knyghty**

### CODE REVIEW

**Comment by knyghty** [comment]
@skorokithakis hmm, I see Travis isn't working here anymore (or I just don't see it). I think you might need to fix something on their side. Or if you like, I can try to find some time to use Github Actions for a CI here...?

**Comment by skorokithakis** [comment]
GH actions would be great, thank you! Also, is there a reason why this PR drops support for older versions?

**Comment by knyghty** [comment]
Per https://www.djangoproject.com/download/ and https://www.python.org/downloads/ they are EOL - I don't see a reason to run the CI for unsupported versions - this package may still work for older versions but I don't see a reason to provide official support for them. I can add back the working ones if you like but I think a more explicit policy to not support EOL versions manages expectations a bit better.

I'll work on moving this to gh actions.

**Comment by knyghty** [comment]
@skorokithakis can you check actions are enabled (in settings -> actions)? Otherwise I might be doing something wrong...

**Comment by knyghty** [comment]
I see the black check isn't working on my repo, I guess it hasn't been kept up to date. I can remove that check, or reformat the files, whichever is better.

**Comment by skorokithakis** [comment]
@knyghty actions are enabled! Let me add a pre-commit config file to this repo so black and everything are kept linted consistently.

**Comment by skorokithakis** [comment]
Actually, pre-commit (and black) works fine for me, did you run it with `pre-commit`?

**Comment by knyghty** [comment]
@skorokithakis as I don't have the thing installed to generate the changelog it nukes it when I try to run the pre-commit hook so I stopped trying to use it. But the black check seems to be not working for me with the same version specified in the pre-commit config. I'm not sure why. You can see the output here: https://github.com/knyghty/django-loginas/pull/1/checks?check_run_id=3252347381

**Comment by knyghty** [comment]
(this last commit removes some code from really old versions of stuff, it'll still work on e.g Python 3.4).

The action doesn't seem to be running still, @skorokithakis - I think for first time contributors you might need to approve the workflow to run - not sure where.

**Comment by skorokithakis** [comment]
That's just because the repo uses 120-char lines and `black` uses 88 by default. The CI needs to run `pre-commit` instead of just some of the checks. I will look into approving the action, hopefully it will be straightforward. Thank you!

**Comment by knyghty** [comment]
I don't think it should make a difference but I'd marked it ready.

black doesn't seem to want to run as part of pre-commit in gh actions - https://github.com/knyghty/django-loginas/pull/1/checks?check_run_id=3252520771 - I'm not sure why. It's also a bit awkward to run pre-commit in there generally... I'm not sure if the pre-commit is supposed to update the changelog but obviously running that as an action isn't going to update the file without commiting/pushing as some dummy user. I think it would probably be better to run the checks separately (changing the default line length for black is simple enough) and have gitchangelog work as an actual pre-commit hook.

**Comment by skorokithakis** [comment]
I've been running `pre-commit` on CI on other projects and it works fine, but you're right, you need to take care to run it with `pre-commit run -a --hook-stage=manual` so it doesn't run `gitchangelog`. I've upgraded all the tools, hopefully that will fix the problem. Can you try?

I think actions don't run because the repo doesn't use an actions file and (I *think*?) the actions file that gets used is the one in the main repo. I will merge your actions when they run and see, I guess? I don't see another way, and I'm not very familiar with actions, I'm afraid...

**Comment by knyghty** [comment]
It should work from this branch, can you try following this? https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

**Comment by knyghty** [comment]
Ah it seems the problem is that the version of black you're using doesn't work with Python 3.9 - it has to be upgraded, which I believe will also cause some reformatting.

File: .github/workflows/ci.yml
Comment (skorokithakis) [inline .github/workflows/ci.yml:18]: I don't think you need to install gitchangelog here.

**Comment by skorokithakis** [comment]
Hmm, the problem is that I don't see the approval button:

image

File: loginas/tests/tests.py
Comment (skorokithakis) [inline loginas/tests/tests.py:17]: Hmm, does this code hurt anything? I'd rather not remove code that doesn't cost anything.

Comment (knyghty) [reply loginas/tests/tests.py:17]: No, but this is just in tests, so if nothing in the test matrix is failing it seems a bit pointless to keep.

This is all quite ancient stuff. `django.urls` was added in 2.0, `urlpase` hasn't been there since Python 2.7, and `importlib` was added back in Python 3.1.

**Comment by knyghty** [comment]
This is all passing in my fork now: https://github.com/knyghty/django-loginas/pull/1 - hopefully the actions will work here once it's merged.

File: loginas/views.py
Comment (skorokithakis) [inline loginas/views.py:12]: Hmm, the tests are fine, but I'd rather not remove this backwards-compatibility code for no reason, even if it's ancient.

Comment (knyghty) [reply loginas/views.py]: Ok, I dropped those commits, though I don't think that any code exists that it will make a difference for. It's still passing on my end.

**Review by skorokithakis — APPROVED** [review]

**Comment by skorokithakis** [comment]
Thank you!

**Comment by knyghty** [comment]
@skorokithakis It looks like the CI is running now, though CircleCI is still trying to run - it probably needs unlinking from the repo. Otherwise, looks like it's working well.

**Comment by skorokithakis** [comment]
I have unlinked CCI, thanks for the help!

**Pull request merged**
